ephios/core/services/notifications/types.py

```diff
@@ -197,12 +201,26 @@ def get_actions(cls, notification):
 
 
 class ParticipationMixin:
+    @classmethod
+    def is_obsolete(cls, notification):
+        participation: AbstractParticipation = AbstractParticipation.objects.get(
+            id=notification.data.get("participation_id")
+        )
+        return participation.state != notification.data["participation_state"]
```

Comment:
What if for the change back another notification was issued. Let's forget about the code complexities for now. We would have two notifications for essentially the same action needed.

Reply:
yeah, but in our discussion we thought about only having one notification per participation. We decided against that to have a proper history of the changes there. So I think it would be weird if we didn't issue two notiifcations only in that scenario

Reply:
I'm worried about the szneario where a notification that has been marked as obsolete becomse unobsolete while at the same time a new notification is created.

You would see these notifications (newest at the top):

* You have been confirmed (unread, not obsolete)
* You have been put to requested (obsolete)
* You have been confirmed (possibly unread, not obsolete as per your code I commented)

I'd expect that last one to be considered obsolete, as there have been newer changes to the participation state, but your code fails to detect these newer changes as the participation is in the original state again.

Reply:
we decided to leave it that way for now and revisit it if users complain or it occurs often